back/app/controllers/web_api/v1/ideas_controller.rb

```diff
@@ -233,6 +246,19 @@ def serialization_options
       }
     end
   end
+
+  def mark_custom_field_values_to_clear!
+    # We need to explicitly mark which custom field values
+    # should be cleared so we can distinguish those from
+    # the custom field value updates cleared out by the
+    # policy (which should stay like before instead of
+    # being cleared out).
+    if @idea&.custom_field_values && params[:idea][:custom_field_values]
+      (@idea.custom_field_values.keys - (params[:idea][:custom_field_values].keys || [])).each do |clear_key|
+        params[:idea][:custom_field_values][clear_key] = nil
+      end
```

Comment:
Is it required for the keys to have a `nil` value? If not, and it is ok to exclude the keys instead, then I think the `each` expression can be refactored to:
```suggestion
      keys_to_clear = @idea.custom_field_values.keys - (params[:idea][:custom_field_values].keys)
      params[:idea][:custom_field_values].except! keys_to_clear
```
Note that I removed the `|| []` part, because `params[:idea][:custom_field_values].keys` cannot return `nil`.

Reply:
Yes, I believe the nil is very much needed to overrite the previous value, while not clearing out the values that were refused by the policy (which could be hidden fields for which we need the data), as indicated by the comment. To be honest I copied this method as is from the users_controller which has the exact sme mechanism of custom fields and refactoring was oos.